### PR TITLE
[DYNAMIC_LINKS] Using correct method to fix tangential passwordless auth issue

### PR DESCRIPTION
**Pull request by PaulTR**
## Description

Fixed passwordless auth by changing getInitialLink in dynamic links (not a valid function name in the Android SDK) back to retrieveDynamicLink (which was correct in previous versions of FlutterFire and currently still used/broken in the sample).

## Related Issues

https://github.com/FirebaseExtended/flutterfire/issues/2642

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [] The analyzer (`flutter analyze`) does not report any problems on my PR.
   Looks like most of this lib is reporting undefined methods?
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by kroikie** [comment]
Seems like `getInitialLink` is used on [iOS](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_dynamic_links/ios/Classes/FLTFirebaseDynamicLinksPlugin.m#L93):

We should update this so that there is consistency across iOS and Android with minimal code change.

**Comment by ditman** [comment]
I think one issue is that the `example` app on firebase_auth  https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth/example/pubspec.yaml#L12 is using a veeery outdated version of the plugin. That dependency should be updated to at least ^0.5.1 to get the latest API available.

Note that in Dart, "^0.3.0" means ">=0.3.0 <0.4.0" and not ">=0.3.0 <1.0.0" (only for versions < than 1.0.0)

File: packages/firebase_dynamic_links/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: firebase_dynamic_links
 description: Flutter plugin for Google Dynamic Links for Firebase, an app solution for creating
   and handling links across multiple platforms.
-version: 0.5.1
+version: 0.6.0
```
Comment (ditman) [inline packages/firebase_dynamic_links/pubspec.yaml:4]: Should be 0.5.2

File: packages/firebase_dynamic_links/lib/src/firebase_dynamic_links.dart [hunk 2]
```diff
@@ -33,7 +33,7 @@ class FirebaseDynamicLinks {
   Future<PendingDynamicLinkData> getInitialLink() async {
     final Map<String, dynamic> linkData =
         await channel.invokeMapMethod<String, dynamic>(
-            'FirebaseDynamicLinks#getInitialLink');
+            'FirebaseDynamicLinks#retrieveDynamicLink');
```
Comment (ditman) [inline packages/firebase_dynamic_links/lib/src/firebase_dynamic_links.dart:36]: I think it'd be easier to just rename the method to `case "FirebaseDynamicLinks#getInitialLink"` in the .java file; that way you shouldn't have to touch this .dart or the iOS plugin, right?

File: packages/firebase_dynamic_links/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.5.2
+
+* Reverted 0.5.0 change that used `getInitialLink()` - a method not available in the Android SDK - back to `retrieveDynamicLink()` specifically for Android
```
Comment (kroikie) [inline packages/firebase_dynamic_links/CHANGELOG.md:3]: Please update the changelog to match.

**Comment by PaulTR** [comment]
Closing while we discuss/work through this

**Pull request closed without merging**